java/src/main/java/com/google/cloud/dataproc/templates/gcs/GCSToSpanner.java

```diff
@@ -54,10 +54,18 @@ public void runTemplate() {
     try (SparkSession spark = SparkSession.builder().appName("GCS to Spanner").getOrCreate()) {
       // Set log level
       spark.sparkContext().setLogLevel(config.getSparkLogLevel());
-
-      Dataset<Row> dataset =
-          spark.read().format(config.getInputFormat()).load(config.getInputLocation());
-
+      Dataset<Row> dataset;
+      if ("csv".equalsIgnoreCase(config.getInputFormat())) {
+        dataset =
+            spark
+                .read()
+                .format(config.getInputFormat())
+                .option("header", "true")
+                .option("inferSchema", "true")
```

Comment:
Here we are infering schema. I think I am okay but we should put in note that spark selected datatypes to be matched with Spanner table data types otherwise there can be error. I mean we are relying on spark that spark will generate perfect schema. Otherwise all good. LGTM!

Reply:
Done